storage/mdadm/include/include.sh

```diff
@@ -0,0 +1,619 @@
+#!/bin/bash
+
+# Copyright (c) 2016 Red Hat, Inc. All rights reserved.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+# Author: Chnaghui Zhong <czhong@redhat.com>
+
+rpm -q mdadm || yum install -y mdadm
+JOURNAL_SUPPORT=0
+info=`mdadm --create --help | grep -o "write-journal"`
+if [ "$info" = "write-journal" ]; then
+        JOURNAL_SUPPORT=1
+fi
+
+#Install fio from upstream
+function install_fio() {
+
+	git_url=git://git.kernel.org/pub/scm/linux/kernel/git/axboe/fio.git
+
+	yum install libaio-devel zlib-devel -y
```

Comment:
Can yum/dnf be chosen based on which one is available instead of blind yum use? Multiple other tests are already doing this check if you grep the repo so you can just reuse that.

Reply:
Here's an example https://github.com/CKI-project/tests-beaker/blob/master/networking/sctp/auth/sockopts/runtest.sh#L30

Reply:
Remove the feature definitions that are not needed temporarily (include.sh is renamed to libmdadm.sh), and will follow the provided examples when needed later.